### PR TITLE
fix: don't remove bucket from selection, if not exists

### LINKED ISSUE #3327: notebooks: Notebook crashes when accessing the queryd_operations bucket in tools
I was playing around in tools, and selected the queryd_operations bucket, and it launches the notebook and then instantly errors with 'an internal error has occurred' and the console shows the stack trace below. This doesn't happen for other buckets, and the queryd_operations bucket can be queried in data explorer.

Also, when I select a new blank notebook, and then look at the list of buckets in the dropdown list, queryd_operations is not there.  Could be related to this: https://github.com/influxdata/ui/issues/839

```
react-dom.production.min.js:198 TypeError: Cannot read properties of undefined (reading 'type')
    at context.tsx:324
    at CardList.tsx:123
    at ia (react-dom.production.min.js:199)
    at Ma (react-dom.production.min.js:218)
    at c (scheduler.production.min.js:11)
    at d (scheduler.production.min.js:12)
    at MessagePort.F.port1.onmessage (scheduler.production.min.js:16)
ra @ react-dom.production.min.js:198
i.componentDidCatch.n.callback @ react-dom.production.min.js:211
Ki @ react-dom.production.min.js:193
Xi @ react-dom.production.min.js:193
La @ react-dom.production.min.js:216
Ba @ react-dom.production.min.js:220
(anonymous) @ react-dom.production.min.js:250
t.unstable_runWithPriority @ scheduler.production.min.js:18
xs @ react-dom.production.min.js:250
As @ react-dom.production.min.js:249
Cs @ react-dom.production.min.js:248
vs @ react-dom.production.min.js:245
Ma @ react-dom.production.min.js:218
c @ scheduler.production.min.js:11
d @ scheduler.production.min.js:12
F.port1.onmessage @ scheduler.production.min.js:16
```

https://influxdb.aws.influxdata.io/notebook/from/bucket/queryd_operations

### PR DESCRIPTION
**Pull request by subirjolly**
Closes #3327 

### BEFORE
BEFORE

### AFTER
AFTER

### CODE REVIEW

**Comment by drdelambre** [comment]
Add the bucket to the list so that the builder doesn't look broken!

**Comment by ui-ci-bot-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 51 | :red_circle:
accessibility | 97 | :green_circle:
seo | 90 | :green_circle:
pwa | null | :hankey:
best-practices | 100 | :green_circle:


[Compare your scores with master branch](https://influxdb.aws.influxdata.io/orgs/844910ece80be8bc/dashboards/073c4b95f1089000?lower=now%28%29%20-%202d&vars%5Blighthouse_score_branch%5D=3462)

File: src/flows/pipes/QueryBuilder/BucketSelector.tsx [hunk 1]
```diff
@@ -3,27 +3,34 @@ import React, {FC, useState, useContext, useEffect} from 'react'
...
-  const {loading, buckets} = useContext(BucketContext)
+  const {loading, buckets, addBucket} = useContext(BucketContext)
 
   const [search, setSearch] = useState('')
 
   const selectBucket = (item?: string): void => {
+    const dataBuckets = !!data.buckets ? [data.buckets[0]] : []
     if (!item) {
-      data.buckets = []
+      data.buckets = dataBuckets
```
Comment (drdelambre) [inline src/flows/pipes/QueryBuilder/BucketSelector.tsx:21]: now you can't deselect any bucket from the list? even by clicking on it?

File: src/flows/context/bucket.scoped.tsx [hunk 2]
```diff
@@ -33,6 +35,18 @@ export const BucketProvider: FC = ({children}) => {
   const {getPanelQueries} = useContext(FlowQueryContext)
   const {id} = useContext(PipeContext)
   const scope = getPanelQueries(id)?.scope ?? {}
+  const controller = useMemo(() => new AbortController(), [])
```
Comment (drdelambre) [inline src/flows/context/bucket.scoped.tsx:38]: i think you want to use a `useRef` here

**Review by drdelambre — APPROVED** [review]

File: src/flows/context/bucket.scoped.tsx (hunk 2 above)
Comment (subirjolly) [reply src/flows/context/bucket.scoped.tsx]: thank you!

**Pull request merged**